antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/misc/IntegerList.kt

```diff
@@ -267,11 +267,11 @@ open class IntegerList {
         }
 
         val originalArray = _data!!.toTypedArray()
-        val copiedArray = Arrays.copyOf(originalArray, newLength)
+        val copiedArray = originalArray.copyOf(newLength)
         for (i in originalArray.size until copiedArray.size) {
             copiedArray[i] = 0
         }
-        _data = copiedArray.toIntArray()
+        _data = copiedArray.filterNotNull().toIntArray()
```

Comment:
mmm, is this intended?

Reply:
Unfortunately there are many things in the code where we use nullable types. This is one of them. We should review those if it's really nullable or not. If it can be casted to Collection<Int> it's most likely shouldn't be Collection<Int?> in the first place.